### PR TITLE
Add attributes to xpu device prop

### PR DESCRIPTION
**Pull request by guangyey**
# Motivation
Add some attributes to `XPUDeviceProp` and expose them via `torch.xpu.get_device_properties` and `torch.xpu.get_device_capability`. They can be used in `torch.compile`  or directly passed to triton to generate more optimized code based on device properties.

# Additional Context
expose the following attributes to `torch.xpu.get_device_properties`：
- `has_fp16` (newly added)
- `has_fp64` (newly added)
- `has_atomic64` (newly added)
- `driver_version`
- `vendor`
- `version`

cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10

### CODE REVIEW

File: torch/csrc/xpu/Module.cpp [hunk 1]
```diff
@@ -227,28 +227,36 @@ static void registerXpuDeviceProperties(PyObject* module) {
...
+                   << prop.driver_version << "', total_memory="
                    << prop.global_mem_size / (1024ull * 1024)
                    << "MB, max_compute_units=" << prop.max_compute_units
                    << ", gpu_eu_count=" << prop.gpu_eu_count
                    << ", gpu_subslice_count=" << gpu_subslice_count(prop)
                    << ", max_work_group_size=" << prop.max_work_group_size
                    << ", max_num_sub_groups=" << prop.max_num_sub_groups
-                   << ", sub_group_sizes=[" << prop.sub_group_sizes << "])";
+                   << ", sub_group_sizes=[" << prop.sub_group_sizes
+                   << "], has_fp64=" << prop.has_fp64 << ")";
```
Comment (jgong5) [inline torch/csrc/xpu/Module.cpp:259]: Why don't you add other new attributes like `has_fp16` and `has_atomic64`?

**Review by jgong5 — CHANGES_REQUESTED** [review]

**Review by EikanWang — COMMENTED** [review]
Overall, LGTM. Please address Jiong's comments.

File: torch/csrc/xpu/Module.cpp (hunk 1 above)
Comment (guangyey) [reply torch/csrc/xpu/Module.cpp]: added.

**Review by jgong5 — APPROVED** [review]

**Comment by guangyey** [comment]
@malfet May I know if you could help review this PR? Thanks very much.

**Comment by guangyey** [comment]
@malfet it would be appreciated if you could help review this PR.

**Review by EikanWang — APPROVED** [review]

**Comment by EikanWang** [comment]
@malfet , @albanD , may I know if you have comments on this PR?

**Comment by guangyey** [comment]
@albanD may I know if you have comments on this PR?

**Review by malfet — APPROVED** [review]

**Review by albanD — APPROVED** [review]
Sounds good.
Any tests?

**Comment by guangyey** [comment]
> Sounds good. Any tests?

We have covered some UT about `get_device_properties` and  `get_device_capability` in test_xpu.py. Will add more tests about `get_device_capability` in this PR.

**Comment by albanD** [comment]
Ho yes I meant for the new properties. Making sure that CI machines are properly detected as what they are.

**Review by atalman — APPROVED** [review]

**Comment by guangyey** [comment]
> Ho yes I meant for the new properties. Making sure that CI machines are properly detected as what they are.

Thanks, I got your point. And add tests in UT.

**Review by albanD — APPROVED** [review]
Thanks!

**Pull request closed without merging**